Scripts/Runtime/Collections/DeferredNativeArray.cs

```diff
@@ -0,0 +1,333 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Collections
+{
+    [BurstCompatible]
+    [StructLayout(LayoutKind.Sequential)]
+    internal unsafe struct DeferredNativeArrayBufferInfo
```

Comment:
I feel like I'm going crazy with this one... 

I agree, they should be internal structs to the container type. But I remember when I did that back with the `UnsafeTypedStream` it would give me a C# error. I can't exactly remember, but it had something to do with unsafe and/or the pointers. 

To remember what that error was, I moved the `BufferInfo` inside the `DeferredNativeArray` but.... no error. 
So I moved all the structs inside the `UnsafeTypedStream` and... also no error. 

Unity (see `UnsafeStream`) keeps them outside and I remember thinking they did that specifically so they could get around the error. 

But seeing as I'm not getting any errors, I see your vote and I get it, but I think we would both prefer them to be inside the collection and not prefixed so I made that change on both collections.